A couple of errors when running `make test`.

I'm receiving a couple of errors when running `make test`. 

1.) It appears this one has been broken for some time, which is odd since I'm pretty diligent about running `make test` when doing work locally and reviewing PRs. I have no other explanation other than to say it's been broken for a while.

The problem is that the example JSON string we use to unmarshal into an Annotation type for the purposes of the test incorrectly has a "type" property instead of a "kind" property. This change to the Annotation schema was made in an [early PR](https://github.com/project-alvarium/alvarium-sdk-go/commit/358c41e3ac4de5fa346c79537ca945f9440ca96c) from July 2021.

To fix the test "type" should be changed to "kind" in the sample JSON string and the test hash value aligned with the new value generated by the test.
```
--- FAIL: TestSignAnnotation (0.00s)
    --- FAIL: TestSignAnnotation/valid_ed25519_signature (0.00s)
        base_test.go:95: 
            	Error Trace:	/home/tsconn/Documents/source/github/public/tsconn23/alvarium-sdk-go/internal/annotators/base_test.go:95
            	Error:      	Not equal: 
            	            	expected: "dafcee0a1844b9c5c0db87252067afc06853afefa14b1711a7c24a6eabc6f6b76b91f8aae2ce54f74b54500dbaf303fbb23dc550e151cfe03cef68ae26a22306"
            	            	actual  : "71819f73d6cb76aae901d08281c158ea9ca2912291d326b6c7118c978d28d522fc28ab3a4006df7af8c8ae0ac643573b35f0c3db9a6153d4b110ce325f974c0c"
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1 +1 @@
            	            	-dafcee0a1844b9c5c0db87252067afc06853afefa14b1711a7c24a6eabc6f6b76b91f8aae2ce54f74b54500dbaf303fbb23dc550e151cfe03cef68ae26a22306
            	            	+71819f73d6cb76aae901d08281c158ea9ca2912291d326b6c7118c978d28d522fc28ab3a4006df7af8c8ae0ac643573b35f0c3db9a6153d4b110ce325f974c0c
            	Test:       	TestSignAnnotation/valid_ed25519_signature
FAIL
```

2.) I suspect this one is due to the commented out logic related to the annotator.Kind validation which we have proposed removing in issue #62 
```
--- FAIL: TestSDKInfo_AnnotationInvalid (0.00s)
    functions.go:36: did not receive expected error: test sdk invalid annotation
FAIL
coverage: 42.6% of statements
FAIL	github.com/project-alvarium/alvarium-sdk-go/pkg/config	0.005s
```

_Originally posted by @tsconn23 in https://github.com/project-alvarium/alvarium-sdk-go/pull/64#pullrequestreview-2136658582_

--- a/internal/annotators/base_test.go
+++ b/internal/annotators/base_test.go
@@ -15,6 +15,8 @@ package annotators
 
 import (
 	"encoding/json"
+	"testing"
+
 	"github.com/project-alvarium/alvarium-sdk-go/internal/hashprovider/md5"
 	"github.com/project-alvarium/alvarium-sdk-go/internal/hashprovider/none"
 	sha2562 "github.com/project-alvarium/alvarium-sdk-go/internal/hashprovider/sha256"
@@ -24,7 +26,6 @@ import (
 	"github.com/project-alvarium/alvarium-sdk-go/pkg/interfaces"
 	"github.com/project-alvarium/alvarium-sdk-go/test"
 	"github.com/stretchr/testify/assert"
-	"testing"
 )
 
 func TestDeriveHash(t *testing.T) {
@@ -68,7 +69,7 @@ func TestSignAnnotation(t *testing.T) {
 	//I'm using a JSON representation here b/c calling the Annotation constructor will populate different ID and Timestamp values each time.
 	//Thus the resulting signature will be different for each test run if I don't use something static.
 	var a contracts.Annotation
-	sample := "{\"id\":\"01F9MS7QVH8Z3KMW757RGFKCBG\",\"key\":\"dummyKey\",\"hash\":\"none\",\"host\":\"ubuntu\",\"type\":\"tpm\",\"timestamp\":\"2021-07-02T18:35:36.561920812-05:00\"}"
+	sample := "{\"id\":\"01F9MS7QVH8Z3KMW757RGFKCBG\",\"key\":\"dummyKey\",\"hash\":\"none\",\"host\":\"ubuntu\",\"tag\":\"e5ec0811a099446d00006f5d53f3b054f6733112\",\"layer\":\"app\",\"kind\":\"tpm\",\"isSatisfied\":false,\"timestamp\":\"2021-07-02T18:35:36.561920812-05:00\"}"
 	json.Unmarshal([]byte(sample), &a)
 
 	tests := []struct {
@@ -80,7 +81,7 @@ func TestSignAnnotation(t *testing.T) {
 		expectError bool
 	}{
 		{"valid ed25519 signature", private, a,
-			"dafcee0a1844b9c5c0db87252067afc06853afefa14b1711a7c24a6eabc6f6b76b91f8aae2ce54f74b54500dbaf303fbb23dc550e151cfe03cef68ae26a22306",
+			"969307e1116e607f0568d354dc156854b0f9977c95779038dc98c1dd177729644470a51e3d647e50a43cb338a803a1406a6feb3ff748586ee3c75847309de802",
 			ed25519.New(), false},
 	}
 	for _, tt := range tests {

--- a/pkg/config/sdk_test.go
+++ b/pkg/config/sdk_test.go
@@ -16,7 +16,6 @@ package config
 
 import (
 	"encoding/json"
-	"github.com/project-alvarium/alvarium-sdk-go/test"
 	"os"
 	"testing"
 )
@@ -34,22 +33,22 @@ func TestSDKInfo_UnmarshalJSON(t *testing.T) {
 	}
 }
 
-func TestSDKInfo_AnnotationInvalid(t *testing.T) {
-	b, err := os.ReadFile("../../test/res/config.json")
-	if err != nil {
-		t.Fatalf(err.Error())
-	}
+// func TestSDKInfo_AnnotationInvalid(t *testing.T) {
+// 	b, err := os.ReadFile("../../test/res/config.json")
+// 	if err != nil {
+// 		t.Fatalf(err.Error())
+// 	}
 
-	var cfg SdkInfo
-	err = json.Unmarshal(b, &cfg)
-	if err != nil {
-		t.Fatalf(err.Error())
-	}
+// 	var cfg SdkInfo
+// 	err = json.Unmarshal(b, &cfg)
+// 	if err != nil {
+// 		t.Fatalf(err.Error())
+// 	}
 
-	cfg.Annotators = append(cfg.Annotators, "invalid")
-	b, _ = json.Marshal(cfg)
+// 	cfg.Annotators = append(cfg.Annotators, "invalid")
+// 	b, _ = json.Marshal(cfg)
 
-	var x SdkInfo
-	err = json.Unmarshal(b, &x)
-	test.CheckError(err, true, "test sdk invalid annotation", t)
-}
+// 	var x SdkInfo
+// 	err = json.Unmarshal(b, &x)
+// 	test.CheckError(err, true, "test sdk invalid annotation", t)
+// }

--- a/pkg/config/sdk_test.go
+++ b/pkg/config/sdk_test.go
@@ -32,23 +32,3 @@ func TestSDKInfo_UnmarshalJSON(t *testing.T) {
 		t.Fatalf(err.Error())
 	}
 }
-
-// func TestSDKInfo_AnnotationInvalid(t *testing.T) {
-// 	b, err := os.ReadFile("../../test/res/config.json")
-// 	if err != nil {
-// 		t.Fatalf(err.Error())
-// 	}
-
-// 	var cfg SdkInfo
-// 	err = json.Unmarshal(b, &cfg)
-// 	if err != nil {
-// 		t.Fatalf(err.Error())
-// 	}
-
-// 	cfg.Annotators = append(cfg.Annotators, "invalid")
-// 	b, _ = json.Marshal(cfg)
-
-// 	var x SdkInfo
-// 	err = json.Unmarshal(b, &x)
-// 	test.CheckError(err, true, "test sdk invalid annotation", t)
-// }

